### PR TITLE
Fix for furnace process timer drift

### PR DESCRIPTION
**Pull request by mcsakoff**
As discussed on #bravo channel we shall protect the furnace process timer from drifting. I have tested couple of implementations and this one works and looks better than others.

### CODE REVIEW

File: bravo/tests/utilities/test_furnace.py [hunk 1]
```diff
@@ -169,3 +171,31 @@ class TestFurnaceProcessCrafting(unittest.TestCase):
...
+            self.assertEqual(self.tile.inventory.crafted[0], (blocks['glass'].slot, 0, 64))
+            headers = [header[0] for header, params in self.protocol.write_packet_calls]
+            # 2 updates for fuel slot (2 saplings burned)
+            # 1 updates for crafting slot (1 sand blocks melted)
+            # 1 updates for crafted slot (1 glass blocks crafted)
+            self.assertEqual(headers.count('window-slot'), 4)
+
+        # smelting time of 2 blocks is ~20s
+        d = deferLater(reactor, 23, done)
+        # but we block the reactor for longer period (12 sec)
```
Comment (mcsakoff) [inline bravo/tests/utilities/test_furnace.py:197]: ops. It's 25s actually...

**Comment by Varriount** [comment]
Erm, won't time.sleep() block the entire reactor? Unless you have the furnace running in a seperate thread (which is almost pointless with loopingcall) or a process (Which should only be used for cpu intensive stuff), you might want to find another way.

**Comment by mcsakoff** [comment]
Are you talking about the test case? The point of the test is to block the application for really long time. ;)

**Comment by MostAwesomeDude** [comment]
I can't believe I'm merging this request. time.sleep() in tests is crazy. The furnace implementation still doesn't actually address the fundamental issue of counting time in seconds instead of ticks. .withCount()'s an improvement, but I'm still not totally sold on it. Worse, there's no documentation as to how the actual implementation works.

The beds and doors are poorly commented and have pretty bad mistakes. Although those mistakes were fixed in subsequent commits, it'd be nice if commits were rebased and cleaned so that I don't have to read through three commits to see that a bug in commit #1 is fixed in commit #3.

The type-based dispatch and non-polymorphic tile handling still sucks. I understand that furnaces are a special case, but not that special.

**Pull request merged**